desktop/core/src/desktop/lib/export_csvxls.py

```diff
@@ -30,7 +30,7 @@
 import sys
 import tablib
 
-from django.http import StreamingHttpResponse, HttpResponse
+from django.http import HttpResponse
```

Comment:
This is fishy to not handle via streaming (and we should still or it could blow-up on big files).

What was the actual root cause of the issue? (to see if they are better way to fix it)

Reply:
@romainr  for transfering the truncated information to UI we use cookie and its set at here : https://github.com/cloudera/hue/blob/3df236d1abb7d4425d0986e2c63cc0c0d9c59b94/desktop/libs/notebook/src/notebook/views.py#L384

and that will be only known to python on whether to truncate or not when the rows are parsed at here : https://github.com/cloudera/hue/blob/541d6597e5e857a758dda89f1b9514889df3c81a/apps/beeswax/src/beeswax/data_export.py#L165-L168 (once we parse the response)

For StreamingHttpResponse this is parseed as streaming so before inhand we set the cookie as truncated=false where UI thinks result is not truncated but infact its truncated.

So we should either change the design of setting the data in cookie or ignore use of streamingHttpResponse, i dont see we are using same in XLS , only for CSV currently streamingHttpResponse is used.

Reply:
Yes, only csv files can be `streamingHttpResponse`, xls creation does not support generators AFAIK. And yes, better to design so that we can keep `streamingHttpResponse` as is for csv!

Reply:
ack, Closing this PR and will raise new One.